processing/src/main/java/org/apache/druid/query/DataSource.java

```diff
@@ -118,6 +123,26 @@ public interface DataSource
    */
   DataSource withUpdatedDataSource(DataSource newSource);
 
+  /**
+   * Returns the query with an updated datasource based on the policy restrictions on tables.
+   * <p>
+   * If this datasource contains no table, no changes should occur.
+   *
+   * @param policyMap a mapping of table names to policy restrictions
```

Comment:
Please add something to the javadoc about the significance of missing-key vs. `Optional.empty` in the map.